test/python/common.py

```diff
@@ -174,4 +174,31 @@ def _is_ci_fork_pull_request():
     return False
 
 
+def compare_dicts(dict1, dict2, threshold, default=0):
```

Comment:
Since it seems the function will be only used in the tests and used directly for making an assertion, can you rename it to `QiskitTestCase.assertAlmostEqualDict()` or similar? This way we can just do `self.assertSomething(counts, target, threshold)` directly, and also ideally get a descriptive message if the assertion fails (instead of returning `False`, it should raise an `AssertionError('....')`.